src/compiler/checker.ts

```diff
@@ -33565,11 +33565,14 @@ namespace ts {
                     case SyntaxKind.SourceFile:
                         return DeclarationSpaces.ExportType | DeclarationSpaces.ExportValue | DeclarationSpaces.ExportNamespace;
                     case SyntaxKind.ExportAssignment:
+                    case SyntaxKind.BinaryExpression:
+                        const node = <ExportAssignment | BinaryExpression>d;

```

Comment:
```suggestion
                        const node = d as ExportAssignment | BinaryExpression;
```

Reply:
@sandersn what do you think about adding `consistent-type-assertions` rule to `eslint`?

Reply:
might be worth a try -- for a while it was 50/50 but by now there might be many more new-style assertions than old-style ones, so the PR might not be *too* large.

Reply:
Do we need to create an issue for this? Or is it a normal open PR without issue?

Reply:
Just a PR -- I'll mark it with the Housekeeping label. (The bot doesn't know about Housekeeping iirc, so it will still probably comment, unfortunately.)